pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/BlockExchange.java

```diff
@@ -137,6 +147,19 @@ protected void sendBlock(SendingMailbox sendingMailbox, TransferableBlock block)
     }
   }
 
+  private void setErrorBlock(TransferableBlock errorBlock) {
+    if (_errorBlock.compareAndSet(null, errorBlock)) {
+      try {
+        for (SendingMailbox sendingMailbox : _sendingMailboxes) {
```

Comment:
Should we call `_callback.accept()` here?

Reply:
doesnt really matter b/c once error is sent. you don't need to reenqueue the opChain (it returned error block and will terminate the opChain)

Reply:
when opChain received an error block (which will be the case when MailboxSendOperator is always the top of an opchain) it will terminate and will not re-enqueue